servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyChannelPublisher.java

```diff
@@ -47,17 +49,21 @@
     private Queue<Object> pending;
     @Nullable
     private Throwable fatalError;
+    @Nullable
+    private NonMultiplexedObserver readWriteObserver;
 
     private final Channel channel;
     private final CloseHandler closeHandler;
     private final EventLoop eventLoop;
     private final Predicate<T> terminalSignalPredicate;
 
-    NettyChannelPublisher(Channel channel, Predicate<T> terminalSignalPredicate, CloseHandler closeHandler) {
+    NettyChannelPublisher(Channel channel, Predicate<T> terminalSignalPredicate, CloseHandler closeHandler,
+                          @Nullable NonMultiplexedObserver readWriteObserver) {
```

Comment:
The `@Nullable` semantics are not ideal here (and in other places) specially here when we can set this later through `readWriteObserver()` method below.

It seems there are two scenarios here:
1. There is no observer configured
2. Observer is configured but is not available because of how our APIs are structured (observer is returned as part of `established()` call)

Instead of wiring in `null` and late set behavior everywhere can we scope it to an observer implementation:
- For case (1) have a NOOP observer
- For case (2) have a place holder observer at the creation time of connection which delegates to an actual observer, when available.

Reply:
There is also a 3rd case when an observer is available at the ctor time for HTTP/2 stream.

Your suggestion to use `when*` operators on `NettyChannelPublisher` helped to simplify it a bit. I will defer further improvements of internal implementation for the follow-up PR.